crates/ra_hir/src/expr.rs

```diff
@@ -969,6 +963,26 @@ where
     }
 }
 
+fn expand_macro_to_expr(
+    db: &impl HirDatabase,
+    macro_call: MacroCallId,
+    args: Option<&ast::TokenTree>,
+) -> Option<TreeArc<ast::Expr>> {
+    let rules = db.macro_def(macro_call.loc(db).def)?;
+
+    let args = mbe::ast_to_token_tree(
+       args?
+    )?.0;
+
+    let expanded = rules.expand(&args).ok()?;
+
+    let result = mbe::token_tree_to_expr(&expanded).ok();
+
+    result
+
+
+}
```

Comment:
extraneous blank lines

Reply:
> Excellent, I think the single thing that is missing now is a smoke test that this work?
> 
> The test should just invoke some trivial macro. Though, we shouldn't use int types in the test, to avoid integer type fallback

I'm trying to add a test for this as a type inference test but I keep on getting a panic. This is because macro expansion produces syntax nodes that dont exist in the current file and as a result it panics with can't resolve local ptr to `SyntaxNode: SyntaxNodePtr { range: [0; 42), kind: BLOCK };` because we look at the syntaxptr for elements that dont `exist` within the current file